File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.go
func leftRightDifference(nums []int) []int {
	n := len(nums)
	ans := make([]int, n)
	leftSum, rightSum := 0, 0
	for i := n - 1; i > 0; i-- {
		rightSum += nums[i]
	}

	for i := 0; i < n; i++ {
		ans[i] = abs(leftSum - rightSum)
		leftSum += nums[i]
		if i < n-1 {
			rightSum -= nums[i+1]
		}
	}

	return ans
}

func abs(n int) int {
	if n < 0 {
		return -n
	}
	return n
}

// array, prefix sum
// time: O(n)
// space: O(1)

File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.py
class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        leftSum, rightSum = 0, 0
        for i in range(len(nums) - 1, 0, -1):
            rightSum += nums[i]
        
        ans = [0] * len(nums)
        for i in range(len(nums)):
            ans[i] = abs(leftSum - rightSum)

            leftSum += nums[i]
            if i < len(nums) - 1:
                rightSum -= nums[i + 1]

        return ans

# array, prefix sum
# time: O(n)
# space: O(1)